File: LV 2/주식가격.py
from collections import deque

def solution(prices):
    answer = [0] * len(prices)
    stack = deque()
    
    for idx in range(len(prices)):
        while stack:
            if prices[stack[-1]] > prices[idx]:
                i = stack.pop()
                answer[i] = idx - i
            else:
                break
                
        stack.append(idx)
    
    while stack:
        i = stack.pop()
        answer[i] = len(prices) - i - 1
        
    return answer

File: LV 2/주식가격.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] prices) {
        int[] answer = new int[prices.length];
        Stack<Integer> stack = new Stack<>();
        int time = 0;

        for (int i = 0; i < prices.length; i++) {
            while (!stack.isEmpty() && prices[stack.peek()] > prices[i]) {
                int removePos = stack.pop();
                answer[removePos] = time - removePos;
            }
            stack.push(i);
            time += 1;
        }

        while (!stack.isEmpty()) {
            int lastTime = prices.length - 1;
            int removePos = stack.pop();
            answer[removePos] = lastTime - removePos;
        }

        return answer;
    }
}
